includes/admin/class-wc-stripe-admin-notices.php

```diff
@@ -240,6 +242,26 @@ public function payment_methods_check_environment() {
 				$this->add_admin_notice( $method, 'notice notice-error', sprintf( __( '%1$s is enabled - it requires store currency to be set to %2$s', 'woocommerce-gateway-stripe' ), $method, implode( ', ', $gateway->get_supported_currency() ) ), true );
 			}
 		}
+
+		if ( ! WC_Stripe_Feature_Flags::is_upe_enabled() || ! WC_Stripe::get_instance()->is_upe_enabled() ) {
+			return;
+		}
+
+		foreach ( WC_Stripe_UPE_Payment_Gateway::UPE_AVAILABLE_METHODS as $method_class ) {
+			if ( WC_Stripe_UPE_Payment_Method_CC::class === $method_class ) {
+				continue;
+			}
+			$method      = $method_class::STRIPE_ID;
+			$show_notice = get_option( 'wc_stripe_show_' . strtolower( $method ) . '_upe_notice' );
+			$upe_method  = new $method_class( null );
+			if ( ! $upe_method->is_enabled() || 'no' === $show_notice ) {
+				continue;
+			}
+			if ( ! in_array( get_woocommerce_currency(), $upe_method->get_supported_currencies() ) ) {
+				/* translators: %1$s Payment method, %2$s List of supported currencies */
+				$this->add_admin_notice( $method . '_upe', 'notice notice-error', sprintf( __( '%1$s is enabled - it requires store currency to be set to %2$s', 'woocommerce-gateway-stripe' ), $method, implode( ', ', $upe_method->get_supported_currencies() ) ), true );
```

Comment:
Do we maybe want to use a better title for the payment method here than the Stripe ID? The current title for existing UPE payment methods, apart from credit/debit cards, appears to be more of a description really. I think it might be useful to add a cleaner, formatted display title to use in cases such as this notice. For instance, `ideal` really should be stylised as `iDEAL`.